backend/src/test/java/com/google/models/DataInfoTest.java

```diff
@@ -0,0 +1,98 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+
+/** Contain tests for methods in {@link DataInfo} and {@link DataInfoUser} class. */
+@RunWith(JUnit4.class)
+public final class DataInfoTest {
+  private static final String METRIC_NAME = "Interest Over Time";
+  private static final String DIMENSION_NAME = "Ramen";
+  private static final String USER_NAME = "catyu@";
+  private static final DataInfo DATA_INFO 
+      = DataInfo.of(METRIC_NAME, DIMENSION_NAME);
+  private static final DataInfoUser DATA_INFO_USER 
+      = DataInfoUser.of(METRIC_NAME, DIMENSION_NAME, USER_NAME);
+
+  @Test
+  public void getMetricName_workingGetter() {
+    assertEquals(METRIC_NAME, DATA_INFO.getMetricName());
+  }
+
+  @Test
+  public void getDimensionName_workingGetter() {
+    assertEquals(DIMENSION_NAME, DATA_INFO.getDimensionName());
+  }
+
+  @Test
+  public void getUsername_workingGetter() {
+    assertEquals(USER_NAME, DATA_INFO_USER.getUsername());
+  }
+
+  @Test
+  public void equals_workingComparatorAndHashCode() {
+    DataInfo same = DataInfo.of(METRIC_NAME, DIMENSION_NAME);
+    DataInfo diffMetric = DataInfo.of("Country", DIMENSION_NAME);
+    DataInfo diffDimension = DataInfo.of(METRIC_NAME, "Udon");
+
+    assertEquals(DATA_INFO, DATA_INFO);
+    assertEquals(DATA_INFO.hashCode(), DATA_INFO.hashCode());
+    assertEquals(DATA_INFO, same);
+    assertEquals(same, DATA_INFO);
+    assertEquals(DATA_INFO.hashCode(), same.hashCode());
+    assertFalse(DATA_INFO.equals(diffMetric));
+    assertFalse(DATA_INFO.hashCode() == diffMetric.hashCode());
+    assertFalse(DATA_INFO.equals(diffDimension));
+    assertFalse(DATA_INFO.hashCode() == diffDimension.hashCode());
+    assertFalse(DATA_INFO.equals(null));
+  }
+
+  @Test
+  public void equals_workingComparatorAndHashCodeChild() {
+    DataInfoUser diffUsername
+        = DataInfoUser.of(METRIC_NAME, DIMENSION_NAME, "leodson@");
+    DataInfoUser diffMetric
+        = DataInfoUser.of("Country", DIMENSION_NAME, USER_NAME);
+
+    // DataInfoUser with different username but same metric and
+    // dimension should be equal. (Could change depending on future
+    // implementation for multiple user per DataInfoUser.)
+    assertEquals(DATA_INFO_USER, DATA_INFO_USER);
+    assertEquals(DATA_INFO_USER.hashCode(), DATA_INFO_USER.hashCode());
+    assertEquals(DATA_INFO_USER, diffUsername);
+    assertEquals(diffUsername, DATA_INFO_USER);
+    assertEquals(DATA_INFO_USER.hashCode(), diffUsername.hashCode());
+    assertFalse(DATA_INFO_USER.equals(diffMetric));
+    assertFalse(diffMetric.equals(DATA_INFO_USER));
+    assertFalse(DATA_INFO_USER.hashCode() == diffMetric.hashCode());
+  }
+
+  @Test
+  public void equals_workingComparatorAndHashCodeBetParentAndChild() {
```

Comment:
does Bet mean Between...?
also just to confirm - this test compares a DataInfo object (the child) with a DataInfoUser object (the parent)?

Reply:
yes lol, I didnt want to make the name very long, maybe i'll just get rid of `Bet` which is confusing

to clarify, `DataInfo` is the parent as `DataInfoUser extends DataInfo` which makes `DataInfoUser` the child. 

yes, this test compares `DataInfoUser` and `DataInfo` to make sure they're equal when the dimension and metric names are the same regardless of whether there are usernames involved or not. This is because I use `DataInfoUser` as key to access a map of `DataInfo` to file locations in order to find related data requested by user, so `DataInfoUser`'s `hashCode` and `equals` method should be same as its parent.

Reply:
ah oops I confused the two. makes sense, thanks for clarifying!